src/main/java/ml/shifu/shifu/udf/BinningPartialDataUDF.java

```diff
@@ -104,8 +104,12 @@ public String exec(Tuple input) throws IOException {
             }
 
             Object value = element.get(1);
-            if(value != null) {
-                binning.addData(value.toString());
+            if (value != null) {
+                if (isWeightBinningMethod() && binning instanceof EqualPopulationBinning) {
```

Comment:
As the formal logical, need further investigation whether to changed for other binning method.